src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -2424,52 +2424,71 @@ class MavlinkStreamGlobalPositionInt : public MavlinkStream
 
 	bool send(const hrt_abstime t) override
 	{
-		vehicle_global_position_s gpos;
 		vehicle_local_position_s lpos;
 
-		if (_gpos_sub.update(&gpos) && _lpos_sub.update(&lpos)) {
-
+		if (_lpos_sub.update(&lpos)) {
 			mavlink_global_position_int_t msg{};
 
+			// time_boot_ms: Timestamp (time since system boot) in ms.
+			msg.time_boot_ms = lpos.timestamp / 1000;
+
+			// alt: Altitude (MSL) in mm.
 			if (lpos.z_valid && lpos.z_global) {
-				msg.alt = (-lpos.z + lpos.ref_alt) * 1000.0f;
+				msg.alt = (-lpos.z + lpos.ref_alt) * 1000.f;
 
 			} else {
 				// fall back to baro altitude
 				vehicle_air_data_s air_data{};
-				_air_data_sub.copy(&air_data);
 
-				if (air_data.timestamp > 0) {
-					msg.alt = air_data.baro_alt_meter * 1000.0f;
+				if (_air_data_sub.copy(&air_data) && (hrt_elapsed_time(&air_data.timestamp) < 10_s)) {
+					msg.alt = air_data.baro_alt_meter * 1000.f;
 				}
 			}
 
+			// relative_alt: Altitude above ground in mm
 			home_position_s home{};
 			_home_sub.copy(&home);
 
 			if ((home.timestamp > 0) && home.valid_alt) {
 				if (lpos.z_valid) {
-					msg.relative_alt = -(lpos.z - home.z) * 1000.0f;
+					msg.relative_alt = -(lpos.z - home.z) * 1000.f;
 
 				} else {
-					msg.relative_alt = msg.alt - (home.alt * 1000.0f);
+					msg.relative_alt = msg.alt - (home.alt * 1000.f);
 				}
 
 			} else {
 				if (lpos.z_valid) {
-					msg.relative_alt = -lpos.z * 1000.0f;
+					msg.relative_alt = -lpos.z * 1000.f;
 				}
 			}
 
-			msg.time_boot_ms = gpos.timestamp / 1000;
-			msg.lat = gpos.lat * 1e7;
-			msg.lon = gpos.lon * 1e7;
+			// lat, lon: Latitude, Longitude in degE7
+			vehicle_global_position_s gpos{};
```

Comment:
So lat/lon are just going to be 0 if they are not available?

Reply:
Correct (once vehicle_local_position is publishing). This is the behavior @DonLakeFlyer requested that already has special handling in QGC for other platforms.

Reply:
I would have preferred explicit handling, but as long as it's lat=lon=0 I think it's fine.

Reply:
Ok, I'll have to add a check in MAVSDK then.

Reply:
What about using `int32_max` for both instead?

Reply:
> What about using int32_max for both instead?

Yes that would work (I also suggested it), but I think it's too late with the significant existing deployed usage. We could still try to get the mavlink spec updated with the explicit INT32_MAX, but it would be in addition to the lat=lon=0 special case.

Reply:
Ok, if it's already deployed then we might as well document 0/0.

Reply:
@LorenzMeier speak up if you are not ok with this!